packages/suite/src/components/suite/NumberInput.tsx

```diff
@@ -211,26 +240,47 @@ export const NumberInput = ({
             // make sure the cursor stays in the right place after formatting
             inputRef.current.setSelectionRange(selectionStart, selectionStart);
         },
-        [formatDisplayValue, locale, onChange, onChangeCallback, inputRef],
+        [formatDisplayValue, locale, onChange, onChangeCallback, inputRef, invalid, rules],
     );
 
     // copy the non-formatted value
     const handleCopy = useCallback(
         (e: React.ClipboardEvent<HTMLInputElement>) => {
-            e.clipboardData.setData('text/plain', cleanValueString(displayValue, locale));
+            if (!inputRef.current) {
+                return;
+            }
+
+            const { selectionStart, selectionEnd, value } = inputRef.current;
+            if (selectionStart === null || selectionEnd == null) {
```

Comment:
`==" is here on purpose?

Reply:
https://github.com/trezor/trezor-suite/pull/7995/commits/148d58e0e8a065a294e2ca4894db4aeb0df76e69